macros/find_all_raw_references.sql

```diff
@@ -0,0 +1,238 @@
+                # from or join followed by at least 1 whitespace character 
+                (from|join)\s+
+
+                # second matching group
+                # 1 or 0 of (opening bracket, backtick, or quotation mark)            
+                ([\[`\"\']?)
+
+                # third matching group
+                # at least 1 word character
+                (\w+)
+                # fouth matching group
+                # 1 or 0 of (closing bracket, backtick, or quotation mark)            
+                ([\]`\"\']?)
+
+                # fifth matching group
+                # a period            
+                (\.)
+
+                # sixth matching group
+                # 1 or 0 of (opening bracket, backtick, or quotation mark)
+                ([\[`\"\']?)
+
+                # seventh matching group
+                # at least 1 word character            
+                (\w+)
+
+                # eighth matching group
+                # 1 or 0 of (closing bracket, backtick, or quotation mark) 
+                ([\]`\"\']?)
+
+                # ninth matching group
+                # a period             
+                (\.)
+
+                # tenth matching group
+                # 1 or 0 of (closing bracket, backtick, or quotation mark)             
+                ([\[`\"\']?)
+
+                # eleventh matching group
+                # at least 1 word character   
+                (\w+)
+
+                # twelfth matching group
+                # 1 or 0 of (closing bracket, backtick, or quotation mark) folowed by a whitespace character or end of string
+                ([\]`\"\']?)(?=\s|$)
+
+                ',
+            'from_table_3':
+                '(?ix)
+
+                # first matching group
+                # from or join followed by at least 1 whitespace character             
+                (from|join)\s+
+
+                # second matching group
+                # 1 or 0 of (opening bracket, backtick, or quotation mark)            
+                ([\[`\"\'])
+
+                # third matching group
+                # at least 1 word character or space 
+                ([\w ]+)
+                
+                # fourth matching group
+                # 1 or 0 of (closing bracket, backtick, or quotation mark) folowed by a whitespace character or end of string
+                ([\]`\"\'])(?=\s|$)
+
+                '
+        } -%}
+
+        {%- for regex_name, regex_pattern in from_raw_references.items() -%}
+
+            {%- set all_regex_matches = re.findall(regex_pattern, model_raw_sql) -%}
+                
+                {%- for match in all_regex_matches -%}
+
+                    {%- set raw_reference = match[1:]|join()|trim -%}
+
+                    {%- do all_raw_references_list.append(raw_reference) -%}
+
+                {%- endfor -%}
+        
+        {%- endfor -%}
+
+    {% endif %}
+
+    # TO DO: consider returning in alphabetic order, and only unique values!
```

Comment:
From version `1.2` dbt has a [`set`](https://docs.getdbt.com/reference/dbt-jinja-functions/set) command (the Python `set`) that can be used to remove duplicates from a list (among other things)